masui/src/main/java/com/ca/mas/ui/MASLoginActivity.java

```diff
@@ -285,7 +292,9 @@ public void onSuccess(MASUser result) {
             public void onError(Throwable e) {
                 progress.dismiss();
                 Toast.makeText(MASLoginActivity.this, "Login failed: " + e.getMessage(), Toast.LENGTH_SHORT).show();
-                MAS.cancelRequest(mRequestId);
+                Bundle data = new Bundle();
+                data.putString("LOGIN_STATUS", "FAILED");
```

Comment:
It is just a way to show the developer how to provide information when canceling the request. It is not retrieved elsewhere. When the MASRequest is built with notifyOnCancel, the provided bundle will be included in RequestCancelledException, and notify the onError callback interface.